### PR TITLE
Update corefx libraries to use xUnit.net 2.1 Beta 2

### PR DESCRIPTION
**Pull request by FiveTimesTheFun**
To support xunit 2.1 Beta 2, update System.Collections.Immutable
to compile against contracts instead of portable targeting packs.

Requires new build tools https://github.com/dotnet/buildtools/pull/171

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@FiveTimesTheFun / @mmitche, I've aborted and restarted the CI build for this a few times, and each time it hangs; appears to be something wrong with the changes, but I've not investigated further to see what.

**Comment by mmitche** [comment]
d:\j\workspace\dotnet_corefx_mac_release_prtest>"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" x86   && build.cmd /p:OSGroup=OSX /p:Configuration=Release /p:SkipTests=true 
  Restoring build tools...
EXEC : error : FindPackagesById: System.Collections [d:\j\workspace\dotnet_corefx_mac_release_prtest\src\Microsoft.CSharp\src\Microsoft.CSharp.csproj]
EXEC : error : FindPackagesById: System.Diagnostics.Debug

**Comment by FiveTimesTheFun** [comment]
These changes depend on corresponding changes in buildtools and a new published buildtools package.  I'll publish the new buildtools package to myget for the CI run.

**Comment by mmitche** [comment]
@FiveTimesTheFun ooh, I think you should pull in my latest build tools changes there too.  Let me know when you've done this.

**Comment by stephentoub** [comment]
@FiveTimesTheFun, @mmitche checked in a change to use new build tools.  What's the next step here?

**Comment by mmitche** [comment]
@stephentoub @FiveTimesTheFun The change to xunit broke buildtools on the CI machines and some other machines.  It introduces a warning about a lack of a strong name for certain assemblies.  This warning is an error on some machines (my laptop, but not desktop, the build machine where the signed bits come from, etc.). I ended up backing how the merge for the time being.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@dotnet-bot test this please.  As an fyi, we're investigating a hang in dnx.exe so I'm rerunning this.

**Comment by jasonwilliams200OK** [comment]
It probably requires `git pull --rebase` followed by merge conflict resolution.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/NuGet.Config [hunk 1]
```diff
@@ -5,6 +5,7 @@
   </packageRestore>
   <packageSources>
     <clear/>
+    <add key="local package source" value="d:\packages" />
```
Comment (mmitche) [inline src/NuGet.Config]: @stephentoub @FiveTimesTheFun I believe this line is causing all of the dnx hangs.

Comment (FiveTimesTheFun) [inline src/NuGet.Config]: Oh geeze... thanks for catching that mmitche.

**Comment by stephentoub** [comment]
@FiveTimesTheFun, I'm going to close this out for now.  We should absolutely move to the newest xunit, but as it is this PR has gone stale (referencing older versions, missing libraries and project files that have been moved to the repo since, etc.), and the necessary NuGet package version referenced by it do not exist.  @FiveTimesTheFun, when you're ready to redo this, please open a new PR. Thanks!

**Pull request closed without merging**